sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -402,6 +403,19 @@ case class DivideYMInterval(
   override def inputTypes: Seq[AbstractDataType] = Seq(YearMonthIntervalType, NumericType)
   override def dataType: DataType = YearMonthIntervalType
 
+  def checkDivideOverflow(month: Int, num: Any): Unit = {
+    if (month == Int.MinValue) {
+      num match {
+        case l: Long if l == -1L => throw QueryExecutionErrors.overflowInIntegralDivideError()
+        case number: Number if number.doubleValue() == -1.0D =>
+          throw QueryExecutionErrors.overflowInIntegralDivideError()
+        case decimal: Decimal if decimal.equals(Decimal.apply(-1)) =>
```

Comment:
Do we really need to handle Double and Decimal. For example, if we divide by -1.0D, we should get the result of the double type which should not overflow. or I am wrong?

```scala
scala> Int.MinValue / -1
res0: Int = -2147483648

scala> Int.MinValue / -1.0D
res1: Double = 2.147483648E9
```
The overflow should happen when we cast double to int.

Reply:
-2147483648 / -1 = -2147483648 is reasonable?

Looks so strange...

Reply:
But here we will get a int value. so it still should overflow

Reply:
Gentle ping @cloud-fan @maropu @MaxGekk @beliefer @wangyum Hope to got more suggestions from your guys about the behavior here

Reply:
I think @MaxGekk 's comment is very clear. The only unreasonable case is `... / -1`, not `... / -1.0`

Reply:
> I think @MaxGekk 's comment is very clear. The only unreasonable case is `... / -1`, not `... / -1.0`

Ok..

Reply:
Updated。